### PR TITLE
feat(arm): add CKV_AZURE_206 to ensure that Storage Accounts use replication

### PR DESCRIPTION
**Pull request by tehila86127**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    We use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the other types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Each prefix should be accompanied by a scope that specifies the targeted framework. If uncertain, use 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

added a new arm policy to ensure that Storage Accounts use replication

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [X] New and existing tests pass locally with my changes

### CODE REVIEW

**Comment by AdamDev** [comment]
Hi @tehila86127 ,
Where do you give this assumption that  Storage Accounts need replication?

**Review by AdamDev — APPROVED** [review]

File: checkov/arm/checks/resource/StorageAccountsUseReplication.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class StorageAccountsUseReplication(BaseResourceValueCheck):
+    def __init__(self) -> None:
+        name = "Ensure that Storage Accounts use replication"
+        id = "CKV_AZURE_206"
+        supported_resources = ("Microsoft.Storage/storageAccounts",)
+        categories = (CheckCategories.BACKUP_AND_RECOVERY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources,)
+
+    def get_inspected_key(self) -> str:
+        return "sku/name"
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/StorageAccountsUseReplication.py:15]: In the check name, we are looking for `Storage Accounts use replication`
But we are checking the `sku/name`, and I don't think it is correct.

Comment (tehila86127) [reply checkov/arm/checks/resource/StorageAccountsUseReplication.py:15]: in the property sku/name every possible value refers to a specific replication type

Comment (ChanochShayner) [reply checkov/arm/checks/resource/StorageAccountsUseReplication.py:15]: @tsmithv11 Can we rely on the SKU name?
Or should this check be a graph check as well?

Comment (tsmithv11) [reply checkov/arm/checks/resource/StorageAccountsUseReplication.py:15]: According to the docs, SKU is good enough: https://learn.microsoft.com/en-us/azure/storage/common/redundancy-migration?tabs=azure-cli

**Review by tsmithv11 — APPROVED** [review]

**Review by ChanochShayner — APPROVED** [review]

**Pull request merged**